code/modules/reagents/reagent_dispenser.dm

```diff
@@ -1,196 +1,340 @@
 /obj/structure/reagent_dispensers
-	name = "Dispenser"
-	desc = "..."
-	icon = 'icons/obj/objects.dmi'
-	icon_state = "water"
+	name = "storage tank"
+	desc = "It can hold large amount of chemicals."
+	icon = 'icons/obj/chemtank.dmi'
+	icon_state = "chemtank"
 	density = 1
 	anchored = 0
 	pressure_resistance = 2*ONE_ATMOSPHERE
-	obj_integrity = 300
-	max_integrity = 300
-	var/tank_volume = 1000 //In units, how much the dispenser can hold
-	var/reagent_id = "water" //The ID of the reagent that the dispenser uses
+	var/list/brokenvars = list("storage tank", "chemtank", /obj/structure/reagent_dispensers)// when it blows up,what will it become,made it qdel src and make a statuelike thing to avoid having to change syringes,droppers,glasses etc with snowflakey checks
+	var/exploded = 0 // if atleast one reaction happened in boom(), this becomes 1
+	var/list/bypassUIicons = list( // list which have all the icons where reagent overlay and update_icon proc in general must not work

```

Comment:
Please do not initialize lists directly on the object, or in `/new` please use the `LAZY_` list helpers when working with lists.

More about that can be found [here](https://github.com/HippieStation/HippieStation13/blob/master/code/__HELPERS/lists.dm#L459)

**ALSO**

`bypassUIicons` is a bad variable name, change this into `lower_case_` also just override `update_icon` and check it in there for these specific objects